app/controllers/ncr/work_orders_controller.rb

```diff
@@ -39,9 +39,12 @@ def update
       @work_order = self.work_order
       @work_order.assign_attributes(permitted_params)   # don't hit db yet
       @approver_email = params[:approver_email]
-
       if self.errors.empty?
-        @work_order.save
+        if @work_order.changed?
+          comment_text = @work_order.changed_attributes_comment
+          @work_order.proposal.changed_fields(comment_text)
+          @work_order.save
+        end
```

Comment:
Can we do this in an `after_update` callback on the model? Just so there's less logic needed in the controller.

http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html